tensorflow_estimator/python/estimator/estimator.py

```diff
@@ -901,7 +902,8 @@ def _export_all_saved_models(
           gfile.MakeDirs(dest_path)
           gfile.Copy(source, dest_absolute)
 
-      gfile.Rename(temp_export_dir, export_dir)
+      if gfile.NeedsTempLocation(export_dir):
+        gfile.Rename(export_dir, export_helpers.get_timestamped_export_dir(export_dir_base))
       return export_dir
 
```

Comment:
Can we add tests for this functionality? Even if mocked, but ideally, there would be some way to know if this is working as intended.